src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,34 +32,45 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        if (address == null) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressDetails = address.split(REGEX_ADDRESS_SPLIT);
+        if (!isValidAddress(addressDetails)) {
```

Comment:
Can you check whether address is null and `isValidAddress` together? This feels like doing the same thing with unnecessary code verbosity.